xchange-acx/src/main/java/org/knowm/xchange/acx/service/trade/AcxTradeService.java

```diff
@@ -91,13 +100,74 @@ public boolean cancelOrder(String orderId) throws IOException {
                 return Stream.of(
                     api.getOrder(accessKey, tonce, Long.valueOf(orderId), signatureCreator));
               } catch (Exception e) {
+
                 throw new RuntimeException("Could not retrieve ACX order with id " + orderId, e);
               }
             })
         .map(order -> mapper.mapOrder(AcxUtils.getCurrencyPair(order.market), order))
         .collect(Collectors.toList());
   }
 
+  @Override
+  public UserTrades getTradeHistory(TradeHistoryParams params) throws IOException {
+    if (!(params instanceof AcxTradeHistoryParams)) {
+      throw new IllegalArgumentException("Invalid tradehistoryparams given");
+    }
+    long tonce = nonceFactory.createValue();
+    AcxTradeHistoryParams acxTradeHistoryParams = (AcxTradeHistoryParams) params;
+    if (acxTradeHistoryParams.currencyPair == null) {
+      throw new IllegalArgumentException("CurrencyPair cannot be null");
+    }
+    Long timestamp = null;
+    Long startId = null;
+    Long endId = null;
+    String order = null;
+    if (acxTradeHistoryParams.endTime != null) {
+      timestamp = acxTradeHistoryParams.endTime.toInstant().getEpochSecond();
+    }
+    String market = getAcxMarket(acxTradeHistoryParams.currencyPair);
+    //    if (acxTradeHistoryParams.endId != null) {
+    //      endId = Long.valueOf(acxTradeHistoryParams.endId);
+    //    }
```

Comment:
why is this commented out?

Reply:
done since it's now using a single method.